.github/workflows/reusable-smoke-test.yml

```
@@ -55,18 +55,27 @@ jobs:
           distribution: temurin
           java-version: 17
 
+      - name: Set up Gradle cache
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: ${{ inputs.cache-read-only }}
```

I think this is still worth limiting github action cache (the other matrix builds will re-use the tomcat one, which is "close"-ish)
```suggestion
          # only push cache for one matrix option per OS since github action cache space is limited
          cache-read-only: ${{ inputs.cache-read-only || matrix.smoke-test-suite != 'tomcat' }}
```